### PR TITLE
Generate type declarations and update typed-error

### LINKED ISSUE #29: Include standalone type declarations

### PR DESCRIPTION
**Pull request by pimterry**
Fixes #29 
Change-type: major (typed-error 3.x requires Node 6+)

### CODE REVIEW

File: build/errors.d.ts [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+ *
+ * @param {String} type - device type
+ * @return {Error} error instance
+ *
+ * @example
+ * throw new errors.BalenaInvalidDeviceType('raspberry-pi')
+ */
+export declare class BalenaInvalidDeviceType extends BalenaError {
+    type: string;
+    constructor(type: string);
```
Comment (Page-) [inline build/errors.d.ts]: You can declare these as `constructor(public type: string);` to avoid all the duplication

Comment (thgreasi) [reply build/errors.d.ts]: Isn't this autogenerated from the ts sources?

Comment (pimterry) [reply build/errors.d.ts]: It is indeed.

Comment (Page-) [reply build/errors.d.ts]: Oh right this is a `.d.ts`, I didn't know we committed then

Comment (pimterry) [reply build/errors.d.ts]: We probably shouldn't be, it's just older projects that still do. We should fix that (here and all sorts of other places too), but I don't really want to get into it before the rename.

**Pull request merged**